samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -137,7 +135,11 @@ def __enter__(self):
                                                       self._debug_args,
                                                       self._debugger_path)
 
-        self._check_docker_connectivity()
+        self._container_manager = self._get_container_manager(self._docker_network,
+                                                              self._skip_pull_image)
+
+        if not self._container_manager.is_docker_reachable:
+            raise InvokeContextException("Running AWS SAM projects locally requires Docker. Have you got it installed?")
```

Comment:
I wonder if we can say Docker > x version, I think that requires some investigation though.

Reply:
@TheSriram  SAM CLI is constrained to some Docker version?

Reply:
Not really, just a way to ask people to upgrade to the latest. There have been some instances where users were using a really old version of docker and had to upgrade. example: to get mounting to work as expected.